### PR TITLE
Add wxRibbonToolBar GetToolByPos and GetToolRect coord functions

### PR DESCRIPTION
**Pull request by gvcallen**
This commit adds the following functions to wxRibbonToolBar:
- wxGetToolByPos accepts two wxCoords (x and y) and returns the opaque tool pointer after a recursive search through all the tools for any tool that contains those coords
- wxGetRect accepts a tool index and returns the rect representing the top left and bottom right points of the relevant tool

These functions were available in wxAuiToolBar but were not available in wxRibbonToolBar, and we found that they were necessary for our project, hence this commit.

### CODE REVIEW

File: interface/wx/ribbon/toolbar.h [hunk 1]
```diff
@@ -307,6 +307,16 @@ class wxRibbonToolBar : public wxRibbonControl
     */
     wxRibbonToolBarToolBase* GetToolByPos(size_t pos)const;
 
+    /**
+    Return the opaque pointer corresponding to the given tool, given specified coordinates.
+
+    @return an opaque pointer, NULL if is not found.
+
+    @since 2.9.4
```
Comment (vadz) [inline interface/wx/ribbon/toolbar.h:315]:
```suggestion
    @since 3.1.5
```

File: src/ribbon/toolbar.cpp [hunk 2]
```diff
@@ -429,6 +429,27 @@ wxRibbonToolBarToolBase* wxRibbonToolBar::GetToolByPos(size_t pos)const
     return NULL;
 }
 
+wxRibbonToolBarToolBase* wxRibbonToolBar::GetToolByPos(wxCoord x, wxCoord y)const
+{
+    size_t group_count = m_groups.GetCount();
+    size_t g, t;
+    for(g = 0; g < group_count; ++g)
```
Comment (vadz) [inline src/ribbon/toolbar.cpp:436]:
```suggestion
    for ( size_t g = 0; g < group_count; ++g )
```

File: src/ribbon/toolbar.cpp [hunk 3]
```diff
@@ -429,6 +429,27 @@ wxRibbonToolBarToolBase* wxRibbonToolBar::GetToolByPos(size_t pos)const
...
+    {
+        wxRibbonToolBarToolGroup* group = m_groups.Item(g);
+        size_t tool_count = group->tools.GetCount();
+        for(t = 0; t < tool_count; ++t)
```
Comment (vadz) [inline src/ribbon/toolbar.cpp:440]:
```suggestion
        for ( size_t t = 0; t < tool_count; ++t )
```

File: src/ribbon/toolbar.cpp [hunk 4]
```diff
@@ -429,6 +429,27 @@ wxRibbonToolBarToolBase* wxRibbonToolBar::GetToolByPos(size_t pos)const
...
+        {
+            wxRibbonToolBarToolBase* tool = group->tools.Item(t);
+            wxRect rect(tool->position.x, tool->position.y, tool->size.GetWidth(), tool->size.GetHeight());
```
Comment (vadz) [inline src/ribbon/toolbar.cpp:443]: I think just this should work:
```suggestion
            wxRect rect(tool->position, tool->size);
```

**Review by vadz — COMMENTED** [review]
Thank you for the additions!

No real comments, just some minor formatting/style improvements (which also apply to the other function even though I didn't repeat them there).

**Comment by gvcallen** [comment]
Thanks for the updates! Was planning on getting to it now after Christmas but appreciate that you did. Kind regards

**Pull request merged**